pkg/agent/controller/policy/policy_controller.go

```diff
@@ -513,16 +519,22 @@ func (r *Reconciler) processPolicyRuleDelete(ruleName string) error {
 		return nil
 	}
 
-	klog.Infof("remove rule %s from datapath", flowKey)
-	if err := r.deletePolicyRuleFromDatapath(flowKey); err != nil {
-		return err
-	}
-
-	r.flowKeyReferenceMap[flowKey].Delete(ruleName)
-	if r.flowKeyReferenceMap[flowKey].Len() == 0 {
+	switch r.flowKeyReferenceMap[flowKey].Len() {
+	case 0:
 		delete(r.flowKeyReferenceMap, flowKey)
-	}
 
+	case 1:
+		if r.flowKeyReferenceMap[flowKey].Has(ruleName) {
+			klog.Infof("remove rule %s from datapath", flowKey)
+			if err := r.deletePolicyRuleFromDatapath(flowKey); err != nil {
+				return err
+			}
+			delete(r.flowKeyReferenceMap, flowKey)
+		}
```

Comment:
if don't have rulename with the flowKey reference, maybe something wrong, add a warning message.

Reply:
done